File: riveredge-backend/src/apps/kuaizhizao/schemas/label_station.py
"""标签工位 schemas（R-16）。"""

from typing import Any, Dict, List, Optional

from pydantic import BaseModel, Field


class LabelModelConfigCreate(BaseModel):
    model_code: Optional[str] = Field(None, max_length=80)
    model_name: str = Field(..., max_length=200)
    template_uuid: str = Field(..., max_length=36)
    template_version: Optional[int] = None
    qty_per_box: int = Field(1, ge=1)
    print_copies: int = Field(1, ge=1)
    device_uuid: Optional[str] = None
    validation_hooks: Optional[List[str]] = None
    policy_config: Optional[Dict[str, Any]] = None
    is_active: bool = True
    remarks: Optional[str] = None


class LabelModelConfigUpdate(BaseModel):
    model_name: Optional[str] = Field(None, max_length=200)
    template_uuid: Optional[str] = Field(None, max_length=36)
    template_version: Optional[int] = None
    qty_per_box: Optional[int] = Field(None, ge=1)
    print_copies: Optional[int] = Field(None, ge=1)
    device_uuid: Optional[str] = None
    validation_hooks: Optional[List[str]] = None
    policy_config: Optional[Dict[str, Any]] = None
    is_active: Optional[bool] = None
    remarks: Optional[str] = None


class LabelStationCreate(BaseModel):
    station_code: Optional[str] = Field(None, max_length=50)
    station_name: str = Field(..., max_length=200)
    model_config_id: Optional[int] = None
    default_mode: str = "work"
    device_uuid: Optional[str] = None
    unlock_requires_password: bool = False
    is_active: bool = True
    remarks: Optional[str] = None


class LabelStationUpdate(BaseModel):
    station_name: Optional[str] = Field(None, max_length=200)
    model_config_id: Optional[int] = None
    default_mode: Optional[str] = None
    device_uuid: Optional[str] = None
    unlock_requires_password: Optional[bool] = None
    is_active: Optional[bool] = None
    remarks: Optional[str] = None


class LabelSessionOpen(BaseModel):
    station_id: int
    mode: Optional[str] = "work"
    model_config_id: Optional[int] = None


class LabelScanRequest(BaseModel):
    session_id: int
    barcode: str = Field(..., min_length=1, max_length=200)
    manual: bool = False


class LabelUnlockRequest(BaseModel):
    session_id: int
    reason: Optional[str] = None
    confirm_password: Optional[str] = None


class LabelLockRequest(BaseModel):
    session_id: int
    reason: str = Field(..., min_length=1, max_length=200)


class LabelUnbindRequest(BaseModel):
    box_id: int
    barcode: Optional[str] = None
    reason: Optional[str] = None


class LabelReprintRequest(BaseModel):
    box_id: int


class LabelCleanupRequest(BaseModel):
    before_at: str = Field(..., description="站点墙钟或可解析业务时刻")


class LabelStationSnapshot(BaseModel):
    session: Dict[str, Any]
    station: Dict[str, Any]
    model: Optional[Dict[str, Any]] = None
    current_box: Optional[Dict[str, Any]] = None
    items: List[Dict[str, Any]] = []

File: riveredge-frontend/src/apps/kuaizhizao/services/label-station.ts
import { apiRequest } from '../../../services/api';

const BASE = '/apps/kuaizhizao/label-station';

export type LabelModelConfig = {
  id: number;
  model_code: string;
  model_name: string;
  template_uuid: string;
  template_version?: number | null;
  qty_per_box: number;
  print_copies: number;
  device_uuid?: string | null;
  validation_hooks?: string[];
  policy_config?: Record<string, unknown>;
  is_active: boolean;
  remarks?: string | null;
};

export type LabelStation = {
  id: number;
  station_code: string;
  station_name: string;
  model_config_id?: number | null;
  default_mode: string;
  device_uuid?: string | null;
  unlock_requires_password: boolean;
  is_active: boolean;
  remarks?: string | null;
};

export type LabelStationSnapshot = {
  session: Record<string, unknown>;
  station: LabelStation;
  model?: LabelModelConfig | null;
  current_box?: Record<string, unknown> | null;
  items?: Record<string, unknown>[];
  last_item?: Record<string, unknown>;
  print_job?: Record<string, unknown>;
};

function unwrapList<T>(res: unknown): { items: T[]; total: number } {
  if (Array.isArray(res)) return { items: res as T[], total: res.length };
  const r = res as Record<string, unknown>;
  const items = (r.items ?? r.data ?? []) as T[];
  return { items: Array.isArray(items) ? items : [], total: Number(r.total ?? items.length) };
}

export const labelStationApi = {
  listModels: async (params?: { active_only?: boolean }) =>
    unwrapList<LabelModelConfig>(await apiRequest(BASE + '/models', { method: 'GET', params })),
  createModel: async (data: Partial<LabelModelConfig>) =>
    (await apiRequest(BASE + '/models', { method: 'POST', data })) as LabelModelConfig,
  updateModel: async (id: number, data: Partial<LabelModelConfig>) =>
    (await apiRequest(`${BASE}/models/${id}`, { method: 'PUT', data })) as LabelModelConfig,
  listStations: async (params?: { active_only?: boolean }) =>
    unwrapList<LabelStation>(await apiRequest(BASE + '/stations', { method: 'GET', params })),
  createStation: async (data: Partial<LabelStation>) =>
    (await apiRequest(BASE + '/stations', { method: 'POST', data })) as LabelStation,
  updateStation: async (id: number, data: Partial<LabelStation>) =>
    (await apiRequest(`${BASE}/stations/${id}`, { method: 'PUT', data })) as LabelStation,
  openSession: async (data: { station_id: number; mode?: string; model_config_id?: number }) =>
    (await apiRequest(BASE + '/sessions/open', { method: 'POST', data })) as LabelStationSnapshot,
  getSession: async (id: number) =>
    (await apiRequest(`${BASE}/sessions/${id}`, { method: 'GET' })) as LabelStationSnapshot,
  closeSession: async (id: number) =>
    (await apiRequest(`${BASE}/sessions/${id}/close`, { method: 'POST' })) as LabelStationSnapshot,
  scan: async (data: { session_id: number; barcode: string; manual?: boolean }) =>
    (await apiRequest(BASE + '/scan', { method: 'POST', data })) as LabelStationSnapshot,
  lock: async (data: { session_id: number; reason: string }) =>
    (await apiRequest(BASE + '/lock', { method: 'POST', data })) as LabelStationSnapshot,
  unlock: async (data: { session_id: number; reason?: string; confirm_password?: string }) =>
    (await apiRequest(BASE + '/unlock', { method: 'POST', data })) as LabelStationSnapshot,
  unbind: async (data: { box_id: number; barcode?: string; reason?: string }) =>
    (await apiRequest(BASE + '/unbind', { method: 'POST', data })) as Record<string, unknown>,
  reprint: async (data: { box_id: number }) =>
    (await apiRequest(BASE + '/reprint', { method: 'POST', data })) as Record<string, unknown>,
  cleanup: async (data: { before_at: string }) =>
    (await apiRequest(BASE + '/cleanup', { method: 'POST', data })) as Record<string, unknown>,
};
